compiler/rustc_mir_transform/src/coroutine/by_move_body.rs

```diff
@@ -23,7 +23,10 @@ impl<'tcx> MirPass<'tcx> for ByMoveBody {
             return;
         };
         let coroutine_ty = body.local_decls[ty::CAPTURE_STRUCT_LOCAL].ty;
-        let ty::Coroutine(_, args) = *coroutine_ty.kind() else { bug!() };
+        if coroutine_ty.references_error() {
+            return;
+        }
+        let ty::Coroutine(_, args) = *coroutine_ty.kind() else { bug!("{body:#?}") };
```

Comment:
If more issues like these arise, we should probably just strip the body of tainted bodies after borrowck

Reply:
wdym strip? the bodies are already empty

Reply:
Not if they were error free before borrowck. I assumed that was the case here